clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -692,9 +692,7 @@ mlir::LogicalResult CIRToLLVMConstantOpLowering::matchAndRewrite(
     // during a pass as long as they don't live past the end of the pass.
     attr = op.getValue();
   } else if (mlir::isa<cir::BoolType>(op.getType())) {
-    int value = (op.getValue() ==
-                 cir::BoolAttr::get(getContext(),
-                                    cir::BoolType::get(getContext()), true));
+    int value = (op.getValue() == cir::BoolAttr::get(getContext(), true));
```

Comment:
Is there a reason we can't just use `int value = mlir::cast<cir::BoolAttr>(op.getValue()).getValue()` here?

Reply:
yeah, probably even nicer is

```
else if (auto ba = mlir::dyn_cast<cir::BoolAttr>(op.getValue())) {
    attr = rewriter.getIntegerAttr(typeConverter->convertType(op.getType()), ba.getValue());
```

Reply:
This entire function might be more nicer in terms of cast to attributes directly, not dispatch based on types.

Also nobody guarantees, that if attribute has `cir::BoolType` it is actually `cir::BoolAttr`,
so the previous code might have returned false even in cases that attribute hold "true" value.

I will look into it in next PR.

Reply:
Fixed as suggested for now. Also in incubator: https://github.com/llvm/clangir/pull/1573